src/formats/onenote.ts

```diff
@@ -0,0 +1,444 @@
+				.setValue(false)
+				.onChange((value) => (this.importIncompatibleAttachments = value))
+			);
+		this.contentArea = this.modal.contentEl.createEl('div');
+		// TODO: Add a setting for importDrawingsOnly when InkML support is complete
+		this.microsoftAccountSetting =
+		new Setting(this.modal.contentEl)
+			.setName('Sign in with your Microsoft Account')
+			.setDesc('You need to sign in in order to import your OneNote data.')
+			.addButton((button) => button
+				.setCta()
+				.setButtonText('Sign in')
+				.onClick(() => {
+					this.graphHelper.openOAuthPage();
+
+					document.addEventListener('graphSignedIn', async () => {
+						const userData: User = await this.graphHelper.requestUrl('https://graph.microsoft.com/v1.0/me');
+						this.microsoftAccountSetting.setDesc(
+							`Signed in as ${userData.displayName} (${userData.mail}). If that's not the correct account, sign in again.`
+						);
+
+						await this.showSectionPickerUI();
+					});
+				})
+			);
+	}
+
+	async showSectionPickerUI() {
+		const sectionsUrl = 'https://graph.microsoft.com/v1.0/me/onenote/notebooks?$expand=sections($select=id,displayName)&$select=id,displayName&$orderby=createdDateTime';
+		const notebooks: Notebook[] = (await this.graphHelper.requestUrl(sectionsUrl)).value;
+
+		this.contentArea.createEl('h3', {
+			text: 'Choose what to import',
+			cls: 'modal-title',
+		});
+
+		notebooks.forEach((notebook) => {
+			this.contentArea.createEl('h5', {
+				text: notebook.displayName!,
+				cls: 'modal-title',
+			});
+
+			let sections: OnenoteSection[] | undefined | null = notebook?.sections;
+			let sectionGroups: SectionGroup[] | undefined | null = notebook?.sectionGroups;
+
+			if (sections) this.createSectionList(sections);
+
+			sectionGroups?.forEach((sectionGroup) => {
+				this.contentArea.createEl('h6', {
+					text: sectionGroup.displayName!,
+				});
+				this.createSectionList(sectionGroup.sections!);
+			});
+		});
+	}
+
+	createSectionList(sections: OnenoteSection[]) {
+		const list = this.contentArea.createEl('ul');
+
+		sections?.forEach((section) => {
+			const listElement = list.createEl('li');
+			let label = listElement.createEl('label');
+			let checkbox = label.createEl('input');
+			checkbox.type = 'checkbox';
+
+			label.appendChild(document.createTextNode(section.displayName!));
+			label.createEl('br');
+
+			// Add/remove a section from this.selectedSections
+			checkbox.addEventListener('change', () => {
+				if (checkbox.checked) this.selectedSections.push(section);
+				else {
+					const index = this.selectedSections.findIndex((sec) => sec.id === section.id);
+					if (index !== -1) {
+						this.selectedSections.splice(index, 1);
+					}
+				}
+			});
+		});
+	}
+
+	async import(progress: ProgressReporter): Promise<void> {
+		let outputFolder = await this.getOutputFolder();
+
+		if (!outputFolder) {
+			new Notice('Please select a location to export to.');
+			return;
+		}
+
+		for (let section of this.selectedSections) {
```

Comment:
It would be nice to have the remaining portion of the importer progress bar work.

What about adding 1 to the remaining for each section that we have left. Then when we get into the section we can add the number of pages in that section too. So by the end the number remaining should be 0, but it will always show some remaining until then.

Currently there's no visual indication of whether it's done importing, or just stalled.